### PR TITLE
Improving Sagemaker module

### PR DESCRIPTION
**Pull request by igorborgest**
Improving Sagemaker module.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by igorborgest** [comment]
image

🚀

File: awswrangler/sagemaker.py [hunk 1]
```diff
@@ -22,34 +22,68 @@ def _parse_path(path):
...
 
         return results
+
+    def get_model(self, job_name: str = None, path: str = None, model_name: str = None) -> Any:
+        """
+        Extract and deserialize a Sagemaker's output model (.tat.gz)
+
+        :param job_name: Sagemaker's job name
+        :param path: S3 path (model.tar.gz path)
+        :param model_name: model name (e.g: )
```
Comment (tuliocasagrande) [inline awswrangler/sagemaker.py]: Missing the `model_name` example. In the case of xgboost, the model_name is `xgboost-model`:
```
:param model_name: model name (e.g: xgboost-model)
```

I'm also wondering how sphinx could indicate that this parameter is optional.

File: awswrangler/sagemaker.py [hunk 2]
```diff
@@ -22,34 +22,68 @@ def _parse_path(path):
...
+        :return:
```
Comment (tuliocasagrande) [inline awswrangler/sagemaker.py]: Missing the return information. Could be something like:
```
:return: The deserialized model, in which the type will depend on the algorithm used
```

File: awswrangler/sagemaker.py [hunk 3]
```diff
@@ -22,34 +22,68 @@ def _parse_path(path):
...
+
+        results: Dict[str, Any] = {}
+        for member in members:
+            logger.debug(f"member: {member.name}")
             f = tar.extractfile(member)
-            file_type = member.name.split(".")[-1]
+            file_type: str = member.name.split(".")[-1]
 
             if (file_type == "pkl") and (f is not None):
                 f = pickle.load(f)
```
Comment (tuliocasagrande) [inline awswrangler/sagemaker.py]: Not every model has a `.pkl` extension. XGBoost for instance is named `xgboost-model`.

**Comment by igorborgest** [comment]
https://github.com/awslabs/aws-data-wrangler/pull/96

File: awswrangler/sagemaker.py (hunk 3 above)
Comment (igorborgest) [reply awswrangler/sagemaker.py]: https://github.com/awslabs/aws-data-wrangler/pull/96

File: awswrangler/sagemaker.py (hunk 2 above)
Comment (igorborgest) [reply awswrangler/sagemaker.py]: https://github.com/awslabs/aws-data-wrangler/pull/96

File: awswrangler/sagemaker.py (hunk 1 above)
Comment (igorborgest) [reply awswrangler/sagemaker.py]: https://github.com/awslabs/aws-data-wrangler/pull/96

**Pull request merged**